build.cake

```diff
@@ -74,59 +72,62 @@ Task("Set-Appveyor-Tag")
 Task("Clean")
     .Does(() =>
 {
-    CleanDirectories(new DirectoryPath[] {outputDirectory, androidDirectory, iosDirectory});
+    CleanDirectory(outputDirectory);
 });
 
-Task("Restore-NuGet-Packages")
+Task("Restore")
     .Does(() =>
 {
-    NuGetRestore("./nunit.runner.sln", new NuGetRestoreSettings {
+    NuGetRestore("./nunit.xamarin.sln", new NuGetRestoreSettings {
         Source = new List<string> {
             "https://www.nuget.org/api/v2/",
             "https://www.myget.org/F/nunit/api/v2"
         },
-        Verbosity = NuGetVerbosity.Quiet 
+        Verbosity = NuGetVerbosity.Quiet
     });
 });
 
 Task("Build")
-    .IsDependentOn("Restore-NuGet-Packages")
+    .IsDependentOn("Restore")
     .Does(() =>
 {
-    MSBuild("./nunit.runner.sln", new MSBuildSettings()
+    MSBuild("./nunit.xamarin.sln", new MSBuildSettings()
         .SetConfiguration(configuration)
         .SetPlatformTarget(PlatformTarget.MSIL)
         .WithProperty("TreatWarningsAsErrors", "true")
         .SetVerbosity(Verbosity.Minimal)
         .SetNodeReuse(false)
+        .UseToolVersion(MSBuildToolVersion.VS2017)
     );
 
-    MSBuild("./src/tests/nunit.runner.tests.uwp/nunit.runner.tests.uwp.csproj", new MSBuildSettings()
+    MSBuild("./tests/nunit.runner.tests.uwp/nunit.runner.tests.uwp.csproj", new MSBuildSettings()
         .SetConfiguration(configuration)
         .SetPlatformTarget(PlatformTarget.x86)
         .WithProperty("TreatWarningsAsErrors", "true")
         .WithProperty("AppxPackageSigningEnabled", "false")
         .SetVerbosity(Verbosity.Minimal)
         .SetNodeReuse(false)
+        .UseToolVersion(MSBuildToolVersion.VS2017)
     );
 });
 
 //////////////////////////////////////////////////////////////////////
 // PACKAGE/PUBLISH
 //////////////////////////////////////////////////////////////////////
 
-Task("Create-NuGet-Packages")
+Task("Package")
     .IsDependentOn("Build")
     .Does(() =>
 {
-    CreateDirectory(outputDirectory);
-
-    NuGetPack("nuget/nunit.runners.xamarin.nuspec", new NuGetPackSettings
-    {
-        Version = version,
-        BasePath = basePath,
-        OutputDirectory = outputDirectory,
-    });        
+    MSBuild("./src/nunit.xamarin/nunit.xamarin.csproj", new MSBuildSettings()
+        .SetConfiguration(configuration)
+        .SetPlatformTarget(PlatformTarget.MSIL)
+        .WithProperty("TreatWarningsAsErrors", "true")
+        .WithTarget("Pack")
+        .SetVerbosity(Verbosity.Minimal)
+        .SetNodeReuse(false)
+        .UseToolVersion(MSBuildToolVersion.VS2017)
+    );
```

Comment:
Do we not need anything platform specific at all anymore? Cool! 😎

Reply:
Should this have version passed into it somehow, so the NuGet packages are created with the 'Appveyor tag version' generated above?

Reply:
Probably a good idea, I will look into it.

Reply:
I've always been curious why we have `.SetNodeReuse(false)`?

Reply:
We have `.setNodeReuse(false)` because one of our projects failed to compile without it way back in the day so it ended up getting copied around as we started using Cake in other projects. I can try dropping it since it probably isn't needed anymore. 😄